src/backends/pytorch/torch_backend.cc

```diff
@@ -0,0 +1,687 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions
+// are met:
+//  * Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of NVIDIA CORPORATION nor the names of its
+//    contributors may be used to endorse or promote products derived
+//    from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+// OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#include "src/backends/pytorch/libtorch_backend.h"
+
+#include <NvInfer.h>
+#include <stdint.h>
+#include "cuda/include/cuda_runtime_api.h"
+#include "src/core/constants.h"
+#include "src/core/logging.h"
+#include "src/core/model_config.h"
+#include "src/core/model_config_utils.h"
+#include "src/core/provider.h"
+#include "src/core/server_status.h"
+
+std::pair<bool, const torch::ScalarType>
+ConvertDataTypeToTorchType(const DataType& dtype)
+{
+  torch::ScalarType type;
+  switch (data_type) {
+    case TYPE_UINT8:
+      type = torch::kByte;
+    case TYPE_INT8:
+      type = torch::kChar;
+    case TYPE_INT16:
+      type = torch::kShort;
+    case TYPE_INT32:
+      type = torch::kInt;
+    case TYPE_INT64:
+      type = torch::kLong;
+    case TYPE_FP16:
+      type = torch::kHalf;
+    case TYPE_FP32:
+      type = torch::kFloat;
+    case TYPE_FP64:
+      type = torch::kDouble;
+    case TYPE_UINT16:
+    case TYPE_UINT32:
+    case TYPE_UINT64:
+    case TYPE_STRING:
+    default:
+        return std::make_pair(false, type);
+  }
+
+  return std::make_pair(true, type);
+}
+
+const std::string
+DataTypeName(const DataType& dtype)
```

Comment:
Don't need to define this function as you can use `DataType_Name()` directly (and you are already using it below)